website/nifti-viewer/index.js

```diff
@@ -0,0 +1,232 @@
+        if (voxelCoords) {
+            updateVoxelCoordinates(view, voxelCoords);
+            handleVoxelClick(voxel3dUnits);
+        }
+    });
+}
+
+// Load and display NIfTI image on a specific element
+function loadAndViewImage(element, imageId, view) {
+    const imageIdObject = ImageId.fromURL(imageId);
+    element.dataset.imageId = imageIdObject.url;
+
+    cornerstone.loadAndCacheImage(imageIdObject.url).then(image => {
+        setupImageViewport(element, image);
+        setupImageTools(element, imageIdObject);
+
+        synchronizer.add(element);
+        addMouseDragListener(element, view);
+
+        element.addEventListener('cornerstonestackscroll', (event) => updateSliceIndex(view, event.detail.newImageIdIndex));
+    }).catch(err => {
+        console.error(`Error loading image for ${view} view:`, err);
+    });
+    element.addEventListener('click', function (event) {
+        //TODO: Update the clicked voxel information.
+        console.log(event)
+    });
+}
+
+// Setup viewport and display the image
+function setupImageViewport(element, image) {
+    const viewport = cornerstone.getDefaultViewportForImage(element, image);
+    cornerstone.displayImage(element, image, viewport);
+    cornerstone.resize(element, true);
+}
+
+// Enable tools and interactions for the displayed image
+function setupImageTools(element, imageIdObject) {
+    const numberOfSlices = cornerstone.metaData.get('multiFrameModule', imageIdObject.url).numberOfFrames;
+    const stack = {
+        currentImageIdIndex: imageIdObject.slice.index,
+        imageIds: Array.from({ length: numberOfSlices }, (_, i) => `nifti:${imageIdObject.filePath}#${imageIdObject.slice.dimension}-${i},t-0`)
+    };
+
+    cornerstoneTools.addStackStateManager(element, ['stack']);
+    cornerstoneTools.addToolState(element, 'stack', stack);
+    cornerstoneTools.mouseInput.enable(element);
+    cornerstoneTools.mouseWheelInput.enable(element);
+    cornerstoneTools.pan.activate(element, 2);
+    cornerstoneTools.stackScrollWheel.activate(element);
+    cornerstoneTools.orientationMarkers.enable(element);
+    cornerstoneTools.stackPrefetch.enable(element);
+    cornerstoneTools.referenceLines.tool.enable(element, synchronizer);
+    cornerstoneTools.crosshairs.enable(element, 1, synchronizer);
+}
+
+// Handle voxel click event
+function handleVoxelClick(currentVoxel) {
+    const [nx, ny, nz, nt] = dims;
+    let [voxelX, voxelY, voxelZ] = currentVoxel;
+
+    voxelX = Math.min(Math.max(Math.round(voxelX), 1), nx) - 1;
+    voxelY = Math.min(Math.max(ny - Math.round(voxelY), 1), ny) - 1;
+    voxelZ = Math.min(Math.max(nz - Math.round(voxelZ), 1), nz) - 1;
+
+    const voxelValues = getVoxelValuesAcrossTime(voxelX, voxelY, voxelZ, nx, ny, nz, nt);
+    updateVoxelCoordinatesDisplay(voxelX + 1, voxelY + 1, voxelZ + 1);
+    plotVoxelData(voxelValues);
+}
+
+// Extract voxel values across all time points
+function getVoxelValuesAcrossTime(x, y, z, nx, ny, nz, nt) {
```

Comment:
This doesn't seem to be correct, select bright parts of images but the value is low. I select dark parts of the image but the value is high. Most likely x, y, and z aren't what you think they are. But also is there no cornerstone based way to do this?

Reply:
It's still not working. See how I have a very bright part selected but the values are quite low. Other areas show up much higher on the plot than this.
I'm guessing it's either image orientation differences or still some indexing problems. I don't see how you're handling the orientation differences, so that's one place to start. What I mean by that is that indexing into the raw data doesn't necessarily equate to the displayed x, y, z. There's a possible axis flip and permutation based on the raw data so it's always displayed in the correct orientation.
<img width="582" alt="image" src="https://github.com/user-attachments/assets/ba6559d4-f7c0-4732-9b8e-72823dc468e2">

Reply:
There is orientation difference and handled it here: https://github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/blob/738b6f1aab7769f5d0135e266d4b2e64df4d0edc/website/nifti-viewer/index.js#L92-L94
I used the complement of y and z due to the axis flip. and the indexing seems accurate according to the images I tested. Can you share the image you're using here so that I can reproduce the issue?

Reply:
I can't share the images because of patient privacy, but you can't count on that transformation always being the same. I opened the same image in yours and a full viewer and I see the images and notations are flipped left/right. Here's a screenshot showing that and some header info.
<img width="1367" alt="image" src="https://github.com/user-attachments/assets/1c250df6-8eff-4e60-8ba8-473eae382ff9">